app/build.gradle

```diff
@@ -61,8 +58,20 @@ dependencies {
     implementation 'androidx.navigation:navigation-fragment:2.3.0'
     implementation 'androidx.navigation:navigation-ui:2.3.0'
     implementation 'de.hdodenhof:circleimageview:3.1.0'
-    // Declare testing dependencies
+
+    // Declare local unit testing dependencies
     testImplementation 'junit:junit:4.12'
+    testImplementation 'androidx.test:core:1.0.0'
+    testImplementation 'org.mockito:mockito-core:1.10.19'
+
+    // Declare instrumented unit tests
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
+    androidTestImplementation 'androidx.test:runner:1.1.0'
+    androidTestImplementation 'androidx.test:rules:1.1.0'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.2'
+    androidTestImplementation 'org.hamcrest:hamcrest-library:1.3'
+    androidTestImplementation 'androidx.test.uiautomator:uiautomator:2.2.0'
```

Comment:
Could i ask what this is for?

Reply:
UI Automator is explained in my description; UI automator is for building UI tests that involve interactions on system and user apps.